src/world/besieged_system.cpp

```diff
@@ -26,15 +26,46 @@ along with this program.  If not, see http://www.gnu.org/licenses/
 #include "map/besieged_data.h"
 #include "world/message_server.h"
 
-BesiegedSystem::BesiegedSystem()
-: sql(std::make_unique<SqlConnection>())
-, besiegedData(std::make_unique<BesiegedData>(sql))
+namespace {
+    // Lockless queue used to perform any internal state changes.
+    // This is necessary due to how the class performs operations both on
+    // the time_server the message_server threads.
+    moodycamel::ConcurrentQueue<std::function<void()>> action_queue;
+}
+
+void process_besieged_actions(const bool& requestExit);
```

Comment:
I wasn't able to put the method into an anonymous namespace and then init the thread with it